suite/tests/linux/clone.c

```diff
@@ -101,24 +103,36 @@ test_thread(bool share_sighand, bool clone_vm, bool use_clone3)
 {
     print("%s(share_sighand %d, clone_vm %d, use_clone3 %d)\n", __FUNCTION__,
           share_sighand, clone_vm, use_clone3);
-    if (use_clone3) {
-#ifdef SYS_clone3
-        child = create_thread_clone3(run_with_exit, &stack, share_sighand, clone_vm);
-#else
-        /* If SYS_clone3 is not defined, we simply use SYS_clone instead, so that
-         * the expected output is the same in both cases.
-         */
-        child = create_thread(run_with_exit, &stack, share_sighand, clone_vm);
-#endif
-    } else
-        child = create_thread(run_with_exit, &stack, share_sighand, clone_vm);
+
+    pid_t (*create_thread_func)(void (*fcn)(void), void **stack, bool share_sighand,
```

Comment:
I'd preserve the comment that we use clone instead of clone3 when clone3 is requested and not available so that the expected output is the same in both cases. Maybe something like:

    /* Use create_thread when clone3 is asked for and not available so that
     * the expected output is the same in all cases.
     */

Reply:
Done